CompilerSource/settings.cpp

```diff
@@ -46,5 +51,60 @@ namespace setting
   bool inherit_objects = 0;  // Determines whether objects should automatically inherit locals and events from their parents
   bool automatic_semicolons = 0; // Determines whether semicolons should automatically be added or if the user wants strict syntax
   COMPLIANCE_LVL compliance_mode = COMPL_STANDARD;
-  string keyword_blacklist = "";
+  std::string keyword_blacklist = "";
+}
+
+CompilerInfo compilerInfo;
+
+void map_ey(ey_data& key, std::map<std::string, std::string>& map) {
+  for (const auto& v : key) {
+    map[toUpper(v.first)] = v.second->data().scalar().toString();
+    std::cout << toUpper(v.first) << "=\""; 
+    std::cout << v.second->data().scalar().toString() << "\"" << std::endl;
+  }
+}
+
+bool load_compiler_ey(std::string fPath) {
+
+  std::cout << std::endl << "Loading compiler ey file: " << fPath << std::endl;
+
+  // Clear old info
+  compilerInfo = CompilerInfo(); 
+
+  std::ifstream compiler_ifstream(fPath.c_str());
+
+  // Bail if error
+  if (!compiler_ifstream.is_open())
+    return false;
+
+  // Parse our compiler data file
+  ey_data compiler_yaml = parse_eyaml(compiler_ifstream, fPath.c_str());
+
+  // Write down our top level ey fields (Note yaml toLowers all fields)
+  compilerInfo.name = compiler_yaml.get("name");
+  compilerInfo.maintainer = compiler_yaml.get("maintainer");
+  compilerInfo.target_platform = compiler_yaml.get("target-platform");
+
+  // Write down grouped fields 
+  const std::map<std::string, std::map<std::string, std::string>&> pairs = {
```

Comment:
lmao... This is so reckless...

To be a little more constructive, have you considered extracting the block below into a function, and just invoking it three times with the contents of these braces?

Also, this would be a fantastic place to use a list of supported tags to check map content and provide useful feedback if other developers have used a key that doesn't exist. But I guess that doesn't have to be done now.